### ISSUE TITLE
Z.root on a negative number raises sigfpe 

### ISSUE THREAD

**Issue by TyOverby** [ocaml/Zarith#60]
Z.root on a negative number raises the sigfpe signal instead of an ocaml exception.
 
```
$ utop 
# Z.(root (of_string "-2") 2);;
Floating point exception (core dumped)
$
```

**Comment by antoinemine** [comment]
Should be fixed by PR #61. 
GMP's doc does not specify this, but the code shows that `root x n` fails for n<=0, and for x<0 when n is even, which is sensible. We now trap these cases and throw an OCaml exception instead of letting `mpn_root` crash us.
Could you confirm this corrects the issue? 
There may be other instances worth tracking in zarith...

**Comment by antoinemine** [comment]
PR merged!
